### ISSUE TITLE
Error when creating a CustomMenuBlock

### ISSUE THREAD

**Issue by tardinha** [silverstripe-australia/silverstripe-ba-sis#2]
I'm getting the following error when adding a new Custom Menu Block
# [User Error] Uncaught LogicException: filter can't be called on an UnsavedRelationList.
## GET /admin/block-admin/Block/EditForm/field/Block/add-multi-class/CustomMenuBlock

Line 319 in /Users/tardinha/www/dpc-hub/framework/model/UnsavedRelationList.php
### Source

``` php
310     public function dataQuery() {
311         throw new LogicException(__FUNCTION__ . " can't be called on an UnsavedRelationList.");
312     }
313 
314     public function exclude() {
315         throw new LogicException(__FUNCTION__ . " can't be called on an UnsavedRelationList.");
316     }
317 
318     public function filter() {
319         throw new LogicException(__FUNCTION__ . " can't be called on an UnsavedRelationList.");
320     }
321 
322     public function getRange($offset, $length) {
323         throw new LogicException(__FUNCTION__ . " can't be called on an UnsavedRelationList.");
324     }
325 
```
### Trace
- UnsavedRelationList->filter(ParentID,0) 
  CustomMenuBlock.php:28
- CustomMenuBlock->getCMSFields() 
  GridFieldDetailForm.php:388
- GridFieldDetailForm_ItemRequest->ItemEditForm(GridField,SS_HTTPRequest) 
  GridFieldDetailForm.php:302
- GridFieldDetailForm_ItemRequest->edit(SS_HTTPRequest) 
  RequestHandler.php:288
- RequestHandler->handleAction(SS_HTTPRequest,edit) 
  RequestHandler.php:200
- RequestHandler->handleRequest(SS_HTTPRequest,DataModel) 
  GridField.php:728
- GridField->handleRequest(SS_HTTPRequest,DataModel) 
  RequestHandler.php:222
- RequestHandler->handleRequest(SS_HTTPRequest,DataModel) 
  RequestHandler.php:222
- RequestHandler->handleRequest(SS_HTTPRequest,DataModel) 
  Controller.php:153
- Controller->handleRequest(SS_HTTPRequest,DataModel) 
  LeftAndMain.php:443
- LeftAndMain->handleRequest(SS_HTTPRequest,DataModel) 
  AdminRootController.php:93
- AdminRootController->handleRequest(SS_HTTPRequest,DataModel) 
  Director.php:370
- Director::handleRequest(SS_HTTPRequest,Session,DataModel) 
  Director.php:153
- Director::direct(/admin/block-admin/Block/EditForm/field/Block/add-multi-class/CustomMenuBlock,DataModel) 
  main.php:177

**Comment by nyeholt** [comment]
Just to confirm the steps to reproduce...
- Create a new blockset
- on the blockset  (once it's been saved) create a new Custom Menu Block from the "Blocks" gridfield
- error

?

**Comment by tardinha** [comment]
Correct.

I can add other block types to the Block set without issue.

I also just tried adding a Custom Menu Block to a page, and got the same error.

**Comment by nyeholt** [comment]
Fixed in 2fd55715ea32237217d773db092c7e23addb7b43
